src/main/java/org/ice4j/ice/harvest/AbstractUdpListener.java

```diff
@@ -316,7 +336,12 @@ private void runInHarvesterThread()
         }
         while (true);
 
-        // TODO we are all done, clean up.
+        //now clean up and exit
+        for (MySocket candidateSocket : sockets.values())
+        {
+            candidateSocket.close();
+        }
+        socket.close();
```

Comment:
AbstractUdpListener.socket.close() is called here, and also in AbstractUdpListener.close(). Is there a reason?

Reply:
I've initially left it out here, but considered that the loop could also be broken due to an unexpected exception. That's why I put the call to `socket.close()` back in. If the socket is already closed, this second invocation is a no-op.